mitodl/mit-learn
# Content File Vector Search Endpoint

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/6296

### Description (What does it do?)
This PR introduces the following:
- an endpoint for searching through contentfile chunks **/api/v0/content_files_vector_search/**
- setting some initial indexes for both learning resource and content file collections
- some code consolidation refactoring


### How can this be tested?
1. checkout this branch
2. rebuild your containers (a pin to onxruntime was re-introduced)
3. restart ceontainers
4. make sure you have learning resources and contentfiles loaded locally
5. pick a few resources ids to embed along with their content and run the management command ```python manage.py generate_embeddings --resource-ids 1,2,3,4```
6. once that completes visit the search endpoint at /api/v0/content_files_vector_search/ and try a few queries (q parameter) and validate the relevant chunks returned.
7. try searching again with some filters (available filters: key, course_number, offered_by, platform, content_feature_type, run_readable_id, resource_readable_id)
8. try also filtering down by the "file_extension" parameter (you can specify multiple such as file_extension=.srt&file_extension=.html)

example response :
https://gist.github.com/shanbady/d87687e062fcfa76e39e47e6b0ad3801

### Additional Context
1. the content of the chunk is returned as "chunk_content" in the record
2. this PR re-introduces a pin to a specific version of onnxruntime. latest version appears to be causing an issue see https://github.com/mitodl/mit-learn/pull/1869#discussion_r1872182014
